src/FirebaseAuth.js

```diff
@@ -222,24 +222,23 @@
      */
     _routerMethodOnAuthPromise: function(rejectIfAuthDataIsNull) {
       var ref = this._ref;
-      var deferred = this._q.defer();
 
-      function callback(authData) {
-        if (authData !== null) {
-          deferred.resolve(authData);
-        } else if (rejectIfAuthDataIsNull) {
-          deferred.reject("AUTH_REQUIRED");
-        } else {
-          deferred.resolve(null);
+      return this._utils.promise(function(resolve,reject){
+        function callback(authData) {
+          if (authData !== null) {
+            resolve(authData);
```

Comment:
Not sure I agree with your premise. Angular promises [are now A++ compliant](https://github.com/angular/angular.js/pull/3535). Part of that spec states that once a promise is resolved or rejected, it's state is fixed permanently - future calls to resolve or reject will not change the promises state. Once you call `resolve(A)`, you can call `resolve(B)` or `reject(E)` as often as you like - the promise will always be resolved with `A`.

Are you possibly thinking about always returning a value from a promise handler? That makes a lot of sense. It allows chaining.

That said - I am totally willing to change it.